lib/timeline/component/ItemSet.js

```diff
@@ -2024,25 +2024,19 @@ ItemSet.prototype._onAddItem = function (event) {
     var scale = this.body.util.getScale();
     var step = this.body.util.getStep();
 
-    var newItemData = {
-      start: snap ? snap(start, scale, step) : start,
-      content: 'new item'
-    };
-
+    var newItemData;
     if (event.type == 'drop') {
-      var itemData = JSON.parse(event.dataTransfer.getData("text"))
-      newItemData.content = itemData.content; // content is required 
-      newItemData.type = itemData.type || 'box';
-      newItemData[this.itemsData._fieldId] = itemData.id || util.randomUUID();
-
-      if (itemData.type == 'range' || (itemData.end && itemData.start)) {
-        if (!itemData.end) {
+      var newItemData = JSON.parse(event.dataTransfer.getData("text"))
+      newItemData.content = newItemData.content ? newItemData.content : 'new item'
+      newItemData.start = newItemData.start ? newItemData.start : (snap ? snap(start, scale, step) : start) 
+      newItemData.type = newItemData.type || 'box';
+      newItemData[this.itemsData._fieldId] = newItemData.id || util.randomUUID();
+
+      if (newItemData.type == 'range' || (newItemData.end && newItemData.start)) {
```

Comment:
Could simplify the conditional case - basically just check for a type of `range` without an `end` date.